cmd/argocd-util/commands/clusters.go

```diff
@@ -0,0 +1,155 @@
+				} else {
+					rbacResources, err = clusterauth.GenerateClusterManagerRBAC(clientset, clusterOpts.SystemNamespace, clusterOpts.Namespaces)
+				}
+				errors.CheckError(err)
+			}
+			if clusterOpts.Name != "" {
+				contextName = clusterOpts.Name
+			}
+			clst := cmdutil.NewCluster(contextName, clusterOpts.Namespaces, conf, managerBearerToken, awsAuthConf, execProviderConf)
+			if clusterOpts.InCluster {
+				clst.Server = common.KubernetesInternalAPIServerAddr
+			}
+			if clusterOpts.Shard >= 0 {
+				clst.Shard = &clusterOpts.Shard
+			}
+
+			namespace, _, err := clientConfig.Namespace()
+			errors.CheckError(err)
+			secret, err := genClusterSecret(clst, namespace, clientset)
+			errors.CheckError(err)
+			secret.Kind = kube.SecretKind
+			secret.APIVersion = "v1"
+			err = cmdutil.PrintResource(secret, outputFormat)
+			errors.CheckError(err)
+			if rbacResources != nil && rbacResources.ServiceAccount != nil {
+				err = cmdutil.PrintResource(rbacResources.ServiceAccount, outputFormat)
+				errors.CheckError(err)
+			}
+			if rbacResources != nil && rbacResources.ClusterRole != nil {
+				err = cmdutil.PrintResource(rbacResources.ClusterRole, outputFormat)
+				errors.CheckError(err)
+			}
+			if rbacResources != nil && rbacResources.ClusterRoleBinding != nil {
+				err = cmdutil.PrintResource(rbacResources.ClusterRoleBinding, outputFormat)
+				errors.CheckError(err)
+			}
+			if rbacResources != nil && rbacResources.Role != nil {
+				err = cmdutil.PrintResource(rbacResources.Role, outputFormat)
+				errors.CheckError(err)
+			}
+			if rbacResources != nil && rbacResources.RoleBinding != nil {
+				err = cmdutil.PrintResource(rbacResources.RoleBinding, outputFormat)
+				errors.CheckError(err)
+			}
+		},
+	}
+	command.PersistentFlags().StringVar(&pathOpts.LoadingRules.ExplicitPath, pathOpts.ExplicitFileFlag, pathOpts.LoadingRules.ExplicitPath, "use a particular kubeconfig file")
+	command.Flags().StringVar(&outputFormat, "o", "yaml", "Output format (yaml|json)")
+	cmdutil.AddClusterFlags(command, &clusterOpts)
+	return command
+}
+
+func genClusterSecret(c *argoappv1.Cluster, namespace string, kubeClientset *kubernetes.Clientset) (*apiv1.Secret, error) {
+	secName, err := db.ServerToSecretName(c.Server)
+	if err != nil {
+		return nil, err
+	}
+	clusterSecret := &apiv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secName,
+			Labels: map[string]string{
+				common.LabelKeySecretType: common.LabelValueSecretTypeCluster,
+			},
+			Annotations: map[string]string{
+				common.AnnotationKeyManagedBy: common.AnnotationValueManagedByArgoCD,
+			},
+		},
+	}
+
+	if err = db.ClusterToSecret(c, clusterSecret); err != nil {
+		return nil, err
+	}
+
+	clusterSecret, err = kubeClientset.CoreV1().Secrets(namespace).Create(context.Background(), clusterSecret, metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}})
```

Comment:
No need to validate secret creation. Even with dry run flag secret creation fails if secret with the same name exist. So user cannot create cluster definition if happens to be connected to the k8s which has same cluster configured.